scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/JobRemoveHandler.java

```diff
@@ -61,55 +61,62 @@ private boolean isInFinishedState(InternalJob job) {
 
     @Override
     public Boolean call() {
-        long start = 0;
+        try {
+            long start = 0;
 
-        if (logger.isInfoEnabled()) {
-            start = System.currentTimeMillis();
-            logger.info("Removing job " + jobId);
-        }
+            if (logger.isInfoEnabled()) {
+                start = System.currentTimeMillis();
+                logger.info("Removing job " + jobId);
+            }
 
-        SchedulerDBManager dbManager = service.getInfrastructure().getDBManager();
+            SchedulerDBManager dbManager = service.getInfrastructure().getDBManager();
 
-        List<InternalJob> jobs = dbManager.loadJobWithTasksIfNotRemoved(jobId);
-        TerminationData terminationData;
+            List<InternalJob> jobs = dbManager.loadJobWithTasksIfNotRemoved(jobId);
+            TerminationData terminationData;
 
-        // if the context is not in sync with the database
-        if (jobs.size() != 1) {
-            terminationData = service.getJobs().removeJob(jobId);
-        } else {
-            // if the job was already finished we just remove it from the context
-            if (isInFinishedState(jobs.get(0))) {
+            // if the context is not in sync with the database
+            if (jobs.size() != 1) {
```

Comment:
why != 1 and not jobs.empty()? what if jobs.size is 2 ? or maybe i don t get the exact logic :D

Reply:
we search for a specific jobid from the database, the loadJobWithTasksIfNotRemoved takes as parameter a variable number of argument list of job ids and return a list of InternalJob. For a single job id we are expecting to receive a single job (and 0 job if the job has been removed from the db by something else).